connect-inject/endpoints_controller_test.go

```diff
@@ -849,6 +851,525 @@ func TestReconcileDeleteEndpoint(t *testing.T) {
+		},
+		"pod not ready, no endpoints need to be reconciled": {
+			agentPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "consul-agent",
+				},
+				Spec: corev1.PodSpec{
+					NodeName: "node-foo",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodReady,
+							Status: corev1.ConditionFalse,
+						},
+					},
+					Phase: corev1.PodRunning,
+				},
+			},
+			existingEndpoints: []*corev1.Endpoints{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "endpoint-1",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									NodeName: toStringPtr("node-foo"),
+								},
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "endpoint-3",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									NodeName: toStringPtr("node-foo"),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedRequests: []ctrl.Request{},
+		},
+		"pod not running, no endpoints need to be reconciled": {
+			agentPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "consul-agent",
+				},
+				Spec: corev1.PodSpec{
+					NodeName: "node-foo",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+					Phase: corev1.PodUnknown,
+				},
+			},
+			existingEndpoints: []*corev1.Endpoints{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "endpoint-1",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									NodeName: toStringPtr("node-foo"),
+								},
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "endpoint-3",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									NodeName: toStringPtr("node-foo"),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedRequests: []ctrl.Request{},
+		},
+	}
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+			logger := logrtest.TestLogger{T: t}
+			s := runtime.NewScheme()
+			s.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Pod{}, &corev1.Endpoints{}, &corev1.EndpointsList{})
+			objects := []runtime.Object{test.agentPod}
+			for _, endpoint := range test.existingEndpoints {
+				objects = append(objects, endpoint)
+			}
+
+			fakeClient := fake.NewFakeClientWithScheme(s, objects...)
+
+			controller := &EndpointsController{
+				Client:  fakeClient,
+				Context: ctx,
+				Scheme:  s,
+				Log:     logger,
+			}
+
+			requests := controller.requestsForRunningAgentPods(handler.MapObject{Meta: test.agentPod})
+			require.ElementsMatch(t, requests, test.expectedRequests)
+		})
+	}
+}
```

Comment:
do you think we need a test case for when pod is not found?

Reply:
done :)